src/spikeinterface/extractors/nwbextractors.py

```diff
@@ -8,37 +8,13 @@
 from spikeinterface.core import BaseRecording, BaseRecordingSegment, BaseSorting, BaseSortingSegment
 from spikeinterface.core.core_tools import define_function_from_class
 
-try:
-    from pynwb import NWBHDF5IO, NWBFile
-    from pynwb.ecephys import ElectricalSeries, ElectrodeGroup
-
-    HAVE_NWB = True
-except ModuleNotFoundError:
-    HAVE_NWB = False
-
-try:
-    import h5py
-
-    HAVE_H5PY = True
-except:
-    HAVE_H5PY = False
-
-try:
-    import fsspec
-
-    HAVE_FSSPEC = True
-except ModuleNotFoundError:
-    HAVE_FSSPEC = False
-
-PathType = Union[str, Path, None]
-
-
-def check_nwb_install():
-    assert HAVE_NWB, NwbRecordingExtractor.installation_mesg
 
-
-def check_fsspec_install():
-    assert HAVE_FSSPEC, "To stream NWB data with fsspec, install fsspec: \n\n pip install fsspec aiohttp requests\n\n"
+def import_lazily():
+    "Makes annotations / typing available lazily"
+    global NWBFile, ElectricalSeries, NWBHDF5IO
+    from pynwb import NWBFile
+    from pynwb.ecephys import ElectricalSeries
+    from pynwb import NWBHDF5IO
```

Comment:
Wouldn't it be better to just have `-> "NWBFile"` instead?

Reply:
This has the marginal benefit of allowing me to navigate (click) on the types but it does add this ugly function. I am on the edge. I will like it to keep it here as an experiment though if you are fine with that.